### PR TITLE
First cut changes for single file.

### PR DESCRIPTION
**Pull request by nitin-deamon**

### CODE REVIEW

File: cmd/zc_traverser_local.go [hunk 1]
```diff
@@ -527,23 +527,85 @@ func (t *localTraverser) Traverse(preprocessor objectMorpher, processor objectPr
...
 				singleFileInfo.ModTime(),
 				singleFileInfo.Size(),
 				noContentProps, // Local MD5s are computed in the STE, and other props don't apply to local files
 				noBlobProps,
 				noMetdata,
 				"", // Local has no such thing as containers
-			),
-			processor,
-		)
-		_, err = getProcessingError(err)
```
Comment (linuxsmiths) [inline cmd/zc_traverser_local.go:545]: why did you remove this?
for correctness we should call it, as per the comments

File: cmd/zc_traverser_local.go [hunk 2]
```diff
@@ -527,23 +527,85 @@ func (t *localTraverser) Traverse(preprocessor objectMorpher, processor objectPr
...
-		return err
+			)
+
+			extendedProp, err := common.GetExtendedProperties(common.CleanLocalPath(t.fullPath))
+			if err != nil {
+				fmt.Printf("GetExtendedProperties for file path[%s] returned error: %v", t.fullPath, err)
```
Comment (linuxsmiths) [inline cmd/zc_traverser_local.go:565]: do you want to change Printf to the logger?

File: cmd/zc_traverser_local.go [hunk 3]
```diff
@@ -527,23 +527,85 @@ func (t *localTraverser) Traverse(preprocessor objectMorpher, processor objectPr
...
+			}
+
+			so.lastChangeTime = extendedProp.CTime()
+			so.inode = extendedProp.INode()
```
Comment (linuxsmiths) [inline cmd/zc_traverser_local.go:569]: if GetExtendedProperties() fails above then we cannot really use extendedProp
Also, since we do not have  reliable ctime and inode should we fallback to TargetCompare or at least to a full copy.

File: cmd/zc_traverser_local.go [hunk 4]
```diff
@@ -527,23 +527,85 @@ func (t *localTraverser) Traverse(preprocessor objectMorpher, processor objectPr
...
+			so.isSingleFile = true
+
+			err = processIfPassedFilters(filters,
+				so,
+				processor,
+			)
+			if err != nil {
+				return err
+			}
```
Comment (linuxsmiths) [inline cmd/zc_traverser_local.go:578]: again, do we need to call getProcessingError()?

File: cmd/zc_traverser_local.go [hunk 5]
```diff
@@ -527,23 +527,85 @@ func (t *localTraverser) Traverse(preprocessor objectMorpher, processor objectPr
...
+
+			// Add the dummy folder entry to folderIndexerMap.
```
Comment (linuxsmiths) [inline cmd/zc_traverser_local.go:580]: pls explain with a good comment the rationale behind these two entries and in general how we are handling the singlefile case

File: cmd/syncComparator.go [hunk 6]
```diff
@@ -282,6 +282,15 @@ func (f *syncDestinationComparator) FinalizeTargetDirectory(relativeDir string,
...
 		}
 
+		/*
+		 * Relative path for singleFile case is "", IndexerMap store folder and it's children.
+		 * For SingleFile we created dummyEntry for folder, this file stored as child.
+		 * This relativePath is w.r.t to that dummy folder.
+		 */
+		if storedObject.isSingleFile {
+			storedObject.relativePath = ""
+		}
```
Comment (linuxsmiths) [inline cmd/syncComparator.go:292]: nit, should we do it right at the beginning, after we get the storedObject

File: cmd/syncComparator.go [hunk 7]
```diff
@@ -512,7 +531,7 @@ func (f *syncDestinationComparator) processIfNecessary(destinationObject StoredO
 			}
 		}
 
-		atomic.AddInt64(&f.sourceFolderIndex.totalSize, -storedObjectSize(sourceObjectInMap))
+		atomic.AddInt64(&f.sourceFolderIndex.totalSize, -size)
```
Comment (linuxsmiths) [inline cmd/syncComparator.go:534]: why did u change this?

File: cmd/zc_traverser_local.go (hunk 1 above)
Comment (nitin-deamon) [reply cmd/zc_traverser_local.go:545]: Added back, why i removed it because it doesn't make sense, that call in case of singlefile. As getProcessingError() check if file is ignored or not. If ignore return err= nil. So if file is ignored and we say transfer successful, it make no sense.

File: cmd/zc_traverser_local.go (hunk 3 above)
Comment (nitin-deamon) [reply cmd/zc_traverser_local.go:569]: This is what we doing in regular path, we can't change mode at this mode, we should fail it here.

File: cmd/zc_traverser_local.go (hunk 4 above)
Comment (nitin-deamon) [reply cmd/zc_traverser_local.go:578]: added, but my argument is above.

File: cmd/syncComparator.go (hunk 7 above)
Comment (nitin-deamon) [reply cmd/syncComparator.go:534]: Need to subtract right size from totalsize.
